### PR TITLE
[pure testing] 

### PR DESCRIPTION
**Pull request by GitPaean**
adding one initPrimaryVariablesEvaluation in updateAndCommunicate function

it should not change running results.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
For more complicated field cases, the changes causes difference, while not revealed by the jenkins test cases.  Closing this testing PR now.

**Comment by GitPaean** [comment]
benchmark please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
benchmark please

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModel_impl.hpp [hunk 1]
```diff
@@ -1492,6 +1492,8 @@ namespace Opm {
             auto& events = this->wellState().well(well->indexOfWell()).events;
             if (events.hasEvent(WellState::event_mask)) {
                 well->updateWellStateWithTarget(ebosSimulator_, this->groupState(), this->wellState(), deferred_logger);
+                well->updatePrimaryVariables(this->wellState(), deferred_logger);
```
Comment (GitPaean) [inline opm/simulators/wells/BlackoilWellModel_impl.hpp:1495]: Basically there is no updating of the primary variables before the `solveWellEquation`. Difficult to see the impact of it.

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1.002    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.975    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.981    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.952    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.88     |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.806    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1614

**Comment by ytelses** [comment]
Benchmark result overview: 




| Test    | Configuration                                     | Relative |
| ------- | ------------------------------------------------- | -------- |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 1        | 1.029    |
| opm-git | OPM Benchmark: flow_mpi_extra - Threads: 8        | 0.979    |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 1        | 0.98     |
| opm-git | OPM Benchmark: flow_mpi_norne - Threads: 8        | 0.955    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 1 | 0.878    |
| opm-git | OPM Benchmark: flow_mpi_norne_4c_msw - Threads: 8 | 0.816    |
* Speed-up = Total time master / Total time pull request. Above 1.0 is an improvement. *



View result details @ https://www.ytelses.com/opm/?page=result&id=1615

**Comment by GitPaean** [comment]
Not sure about the testing results, while the change should go in from the code level. Will create a new PR to prepare the merging.

**Pull request closed without merging**